### DISCUSSION TITLE
Will not turn on after sleep

### DISCUSSION THREAD

**Discussion by kusmi** [NikolaiRadke/Schimpfolino#6] [General]
I like this project!

But I run into one problem: after it goes to sleep after some inactivity time, it does not respond to any button presses anymore. Also removing the battery does not help - I guess there is still some electricity in the attiny?

Perhaps I forgot to set some fuses or configuration, so it can be waked up?

**Comment by kusmi** [comment]
ok, when I press a few dozen of times the Schimpfolino sometimes comes back to live and I can then repeatedly push the button and it works as it should. But then again: after it goes to sleep, it has hard time getting back to life again.

I'm using "new" sketch and newest Arduino.

EDIT: Also tried with original Schmipfolino sketch, same behavior. Also tried to set different fuse settings, but had no effect when I "burned" the bootloader.

**Comment by NikolaiRadke** [comment]
Hi,

there seem to be interrupt issues with some ATtinys, maybe china clones. Some other users share the problem, but most don't.
All my own Tinys are working, so i can't reproduce the issue.

What happens if you bypass the button and shorten the connection from the pin to ground?


Best wishes,

Nikolai

**Comment by kusmi** [comment]
Well, I got those ATTinys from AliExpress, they still have ATMEL on the chip (Microchip bought them 2016) so this could be true... Also shorting PB1 (Pin6) to ground has the same problem: only after a dozen or so "shorts" the Schimpfolino comes to live.

I will order from more respectable sources then. Bit of a bummer, as I planned a family build, so we will not run out of swear words during the festive season 😄 

EDIT: OK, found a distributor in Switzerland - will be shipped by December 24 - let's start the swearing season! 😄

**Reply by NikolaiRadke** [reply]
Thumbs are pressed :-)
Christmas must not be to harmonius!

**Comment by kusmi** [comment]
Bummer, I got my ATTiny85s from Distrelec.ch (a local RS-Online company), so they should not be fake - but I have the same problem: After sleep I need to press a few dozen times to wake it up again.

I will now try a different OLED screen, perhaps it wakes up fine, but did not bring the display up?

So even without swear-words-generator I keep swearing :-) hehe

**Comment by kusmi** [comment]
OK, I found my issue: If I increase the **_delay_ms(5)** in in the **Oled_init()** method, It seems to work!

I had to increase it to at least 50 ms: **_delay_ms(50)** - and it now even works with the AliExpress Attiny85s 😄

**Reply by NikolaiRadke** [reply]
That's the solution??? 
I would never have found out.

Whats the value now? I will change it in all sketches.

**Comment by kusmi** [comment]
I set it to 50 ms - 40 ms seems to be the sweetspot when it starts working.

**Reply by NikolaiRadke** [reply]
Thanks! 
Militant Christmas, you are well prepared with mighty weapons!

**Comment by kusmi** [comment]
Full success - what a nice Christmas project :-)
![IMG_6015](https://github.com/user-attachments/assets/f4f11233-6b43-4c6e-a9d6-bcbd90e79443)

**Reply by NikolaiRadke** [reply]
Wow!
I need your case! NOW!!!

**Comment by kusmi** [comment]
Sure - but it was my first FreeCAD project, so take it as it is 😅.

The buttons are from AliExpress:
NO LIGHT, 12mm Flat Head, 3-6V LED, Stainless Shell or Black Shell
https://www.aliexpress.us/item/1005003808215595.html
The button are just beneath the PCB and I soldered some small wires on (just make sure to first screw the button in and only then solder it to PCB)

The screen-cover are 30 mm acrylic glass (In previous projects my kids easily destroyed the finicky OLED displays, so this is just for precaution 😉) - they come handy with protective sheets, so you can easily paint over them and afterwards remove the sheets.
30 mm x 1 mm
https://www.aliexpress.us/item/3256805320017166.html

Then I use normal "Senkkopfschrauben" (that would also be a good swear word :-) to screw the backplate on: 2 mm x 12 mm from the local hardware store.

And I used some 2-component epoxy to glue the screen-cover to the PLA body (and printed special inlay piece to cover the rectangle hole from the display - also I wanted to have a smooth surface and when printed with support it came out ugly. So I printed the main case upside-down, then placed the acrylic cover in the hole, glued the PLA inlay on top and smeared some more glue on hole to cover the inlay) - in my first prototype I did not cover the hole inlay and it looked ugly, so for the kids I made sure I did cover the whole inlay with epoxy...

I also printed some spacer to separate the OLED display from the main PCB - this is not needed, but I did not have thick double-sided tape at hand...

And we used metallic spray cans from the hardware store to give them some "glitter" 😄 

Here are the files:

[Schimpfolino Case and STL.zip](https://github.com/user-attachments/files/18262273/Schimpfolino.Case.and.STL.zip)


![IMG_6023](https://github.com/user-attachments/assets/ab8c9b97-53c3-4e9e-8205-7ef8e16ec7a3)
![IMG_6024](https://github.com/user-attachments/assets/d465b040-2396-4805-a6b2-6dc55de61c77)

**Reply by NikolaiRadke** [reply]
Extremly cool! I will try it out and add a manual for the german Wiki. Thanks for sharing!

**Reply by kusmi** [reply]
Please adapt it as you like - I call it "the boring case" as it is not really monster like...

And perhaps it is tailored to my 3d printer (a very old one), e.g. even if I specify the screw holes to be 3 mm, the holes will be effectively only 1.5 - max 2 mm diameter and fit my screws...
